engine/src/gfxlib_struct.h

```diff
@@ -646,6 +646,36 @@ class GFXLight
               float exp = 0.0f,
               float cutoff = 180.0f,
               float size = 0.0f );
+    
+    GFXLight(const GFXLight& other)
+    {
+        vect[0] = other.vect[0];
+        vect[1] = other.vect[1];
+        vect[2] = other.vect[2];
```

Comment:
A series of for loops for these would be good for simplicity; this works for now so not blocking.

Reply:
I did consider for loops. Unfortunately, they take up more vertical space and requires the same amount of typing as the solution I went for here. Also, for the compiler for loops would require counters and more steps. From an optimization point of view what I have done is a manual loop unroll. 

In Lisp you would iterate over the names of the lists as well as the lists themselves, or have some sort of macro. 

Fortunately, this problem is too simple for such a complicated solution.